crates/std_detect/src/detect/cache.rs

```diff
@@ -63,6 +69,19 @@ impl Initializer {
         let v = self.0;
         self.0 = set_bit(v, bit);
     }
+
+    /// Unsets the `bit` of the cache.
+    #[inline]
+    pub(crate) fn unset(&mut self, bit: u32) {
+        // FIXME: this way of making sure that the cache is large enough is
+        // brittle.
+        debug_assert!(
```

Comment:
I think this will detect errors here in CI, which is where we need to catch these issues and where we try to test all features, so I think this is fine. 

Did you have in mind any other way to improve this ?

Reply:
Not right now, I copied the FIXME from the other functions, we could test this in the macro even if I'm afraid wouldn't be that fun to implement it.

Reply:
I'd just remove the FIXMEs and just replace this with a comment saying what this does (catch if we have outgrown the cache during testing or so).

Reply:
Added a commit about it